cpp/validate_harvested_records.cc

```diff
@@ -76,35 +77,85 @@ struct FieldInfo {
-                         const bool not_in_database_yet): zeder_id_(zeder_id), zeder_instance_(zeder_instance),
-                         not_in_database_yet_(not_in_database_yet) { }
-    JournalInfo() = default;
-    JournalInfo(const JournalInfo &rhs) = default;
+
+    GeneralInfo() = default;
+    GeneralInfo(const GeneralInfo &rhs) = default;
 
     size_t size() const { return field_infos_.size(); }
-    const std::string &getZederId() const { return zeder_id_; }
-    const std::string &getZederInstance() const { return zeder_instance_; }
-    bool isInDatabase() const { return not not_in_database_yet_; }
+
     void addField(const std::string &field_name, const FieldPresence field_presence)
         { field_infos_.emplace_back(field_name, field_presence); }
+
+    void addField(const FieldInfo &field_info) { field_infos_.emplace_back(field_info); }
+
     const_iterator begin() const { return field_infos_.cbegin(); }
     const_iterator end() const { return field_infos_.cend(); }
+    const_iterator find(const std::string &field_name) const {
+        return std::find_if(field_infos_.begin(), field_infos_.end(),
+                            [&field_name](const FieldInfo &field_info){ return field_name == field_info.name_; });
+    }
     iterator begin() { return field_infos_.begin(); }
     iterator end() { return field_infos_.end(); }
     iterator find(const std::string &field_name) {
         return std::find_if(field_infos_.begin(), field_infos_.end(),
                             [&field_name](const FieldInfo &field_info){ return field_name == field_info.name_; });
     }
+
+    // Combine GeneralInfo with other General Info (e.g. JournalInfo).
+    // rhs will have priority to simulate data inheritance.
+    static GeneralInfo Combine(const GeneralInfo &lhs, const GeneralInfo &rhs) {
+        auto lhs_iter(lhs.begin());
+        auto rhs_iter(rhs.begin());
+
+        GeneralInfo combined_info;
+        while (lhs_iter != lhs.end() && rhs_iter != rhs.end()) {
+            if (lhs_iter == lhs.end()) {
+                combined_info.addField(*rhs_iter);
+                ++rhs_iter;
+            } else if (rhs_iter == rhs.end()) {
+                combined_info.addField(*lhs_iter);
+                ++lhs_iter;
+            } else {
+                int compare_result(StringUtil::AlphaWordCompare(lhs_iter->name_, rhs_iter->name_));
+                if (compare_result == 0) {
+                    // if present on both sides, rhs wins!
+                    combined_info.addField(*rhs_iter);
+                    ++lhs_iter;
+                    ++rhs_iter;
+                } else if (compare_result < 0) {
+                    combined_info.addField(*lhs_iter);
+                    ++lhs_iter;
+                } else if (compare_result > 0) {
+                    combined_info.addField(*rhs_iter);
+                    ++rhs_iter;
+                }
+            }
+        }
+
+        return combined_info;
+    }
+};
```

Comment:
Function is way too long to be part of the class definition!

Reply:
done